plugins/org.python.pydev/tests_analysis/com/python/pydev/analysis/refactoring/quick_fixes/AddTokenAndImportStatementTest.java

```diff
@@ -0,0 +1,235 @@
+package com.python.pydev.analysis.refactoring.quick_fixes;
+
+import org.eclipse.jface.text.Document;
+import org.eclipse.text.edits.ReplaceEdit;
+import org.python.pydev.core.log.Log;
+
+import com.python.pydev.analysis.refactoring.quick_fixes.AddTokenAndImportStatement.ComputedInfo;
+
+import junit.framework.TestCase;
+
+public class AddTokenAndImportStatementTest extends TestCase {
+
+    public String checkImport(String test_document, String test_realImportRep, int test_maxCols) throws Exception {
+
+        Document document = new Document(test_document);
+
+        char trigger = '\n';
+        int offset = 0;
+        boolean addLocalImport = false;
+        boolean addLocalImportsOnTopOfMethod = false;
+        boolean groupImports = true;
+        int maxCols = test_maxCols;
+
+        AddTokenAndImportStatement stmt = new AddTokenAndImportStatement(document, trigger, offset, addLocalImport,
+                addLocalImportsOnTopOfMethod,
+                groupImports, maxCols);
+
+        String realImportRep = test_realImportRep;
+        int fReplacementOffset = document.getLineLength(0);
+        int fLen = 0;
+        String indentString = "";
+        String fReplacementString = "";
+        boolean appliedWithTrigger = false;
+        int importLen = 0;
+
+        ComputedInfo computedInfo = new ComputedInfo(realImportRep, fReplacementOffset, fLen, indentString,
+                fReplacementString, appliedWithTrigger, importLen, document);
+        stmt.createTextEdit(computedInfo);
+
+        for (ReplaceEdit edit : computedInfo.replaceEdit) {
+            try {
+                edit.apply(document);
+            } catch (Exception e) {
+                Log.log(e);
+            }
+        }
+
+        return document.get();
+
+    }
+
+    // normal test
+    public void test1() throws Exception {
+        String test_document = "from math import ceil";
+        String test_realImportRep = "from math import sqrt";
+        String doc = checkImport(test_document, test_realImportRep, 80);
+
+        String expected = "from math import ceil, sqrt";
```

Comment:
In cases with many params, having variables extracted may help, but for this case, just doing `assertEquals("from math import ceil, sqrt", doc)` is easier to follow, so, please inline this vars (same for all other test cases here)... you can inline putting the cursor over the variable and then do: `Shift+Alt+i`.

-- actually, seeing the whole test, I think you should have the function do the compare instead of repeating it everywhere, so, the function would be something as:

`public String checkImport(String baseDoc, String importToAdd, String expectedDoc, int maxCols)` -- note how vars don't need to be prefixed by `test` as it's just noise as we're already in a test case (to rename a var you can do put the cursor over the variable and do a `Ctrl+2 and then R`).

`checkImport("from math import sqrt", "from math import ceil, sqrt", "from math import ceil, sqrt", 80);`